Bucketlist/src/app/login/login.component.ts

```diff
@@ -1,23 +1,41 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { Router, ActivatedRoute } from '@angular/router';
 import { AuthenticationService } from '../services/authentication.service';
 import { AlertService } from '../services/alert.service';
 
 @Component({
     selector:'login-app',
     templateUrl:'./login.component.html',
-    styleUrls: ['./login.component.css']
+    styleUrls: ['./login.component.css'],
+    providers:[ AlertService, AuthenticationService ]
 
 })
-export class LoginComponent{
-    model: any;
+export class LoginComponent implements OnInit {
+    loading = false;
     email: string;
     password: string;
     returnUrl: string;
 
-    constructor(
-        private route: ActivatedRoute,
+    constructor(private authService: AuthenticationService,
+                private _service: AlertService,
```

Comment:
Good implementation and naming of private variables ie. with an underscore before the variable